components/mpas-ocean/src/analysis_members/mpas_ocn_harmonic_analysis.F

```diff
@@ -788,24 +791,27 @@ subroutine least_squares_solve(nfreq,ha,hap,hax) !{{{
 
 !     solve for y by forward substitution for l*y=p
 
-      do 120 ir=1,mm
+      do ir=1,mm
          y(ir)=hap(ir)
-         do 110 jr=1,ir-1
- 110        y(ir)=y(ir)-ha(jr,ir)*y(jr)
- 120  continue
+         do jr=1,ir-1
+            y(ir)=y(ir)-ha(jr,ir)*y(jr)
+         end do
+      end do
 
 !     calculate c=d**(-1)*y
 
-      do 130 ir=1,mm
- 130     c(ir)=y(ir)/ha(ir,ir)
+      do ir=1,mm
+         c(ir)=y(ir)/ha(ir,ir)
+      end do
 
 !     solve for x by back-substituting for l(tr)*x=c
 
       ir=mm
  140  continue
       hax(ir)=c(ir)
-      do 150 jr=ir+1,mm
- 150     hax(ir)=hax(ir)-ha(ir,jr)*hax(jr)
+      do jr=ir+1,mm
+         hax(ir)=hax(ir)-ha(ir,jr)*hax(jr)
+      end do
       ir=ir-1
       if(ir.ge.1) goto 140
```

Comment:
do we want to make this into a `do` backward loop?

Reply:
Yes, this is a backward loop by hand! I didn't see that the first time. I'll change it and you can check me.

Reply:
corrected.